src/Storages/MergeTree/ReplicatedMergeTreeTableMetadata.cpp

```diff
@@ -121,6 +150,19 @@ void ReplicatedMergeTreeTableMetadata::write(WriteBuffer & out) const
 
     if (!constraints.empty())
         out << "constraints: " << constraints << "\n";
+
+    if (merge_params_version >= REPLICATED_MERGE_TREE_METADATA_WITH_ALL_MERGE_PARAMETERS)
+    {
+        out << "merge parameters format version: " << merge_params_version << "\n";
```

Comment:
Let's add a compatibility setting that disables the new version. Otherwise, it will be impossible to downgrade after you create a new table or add a new replica. See https://github.com/ClickHouse/ClickHouse/pull/42869#issuecomment-1311785870 
We will enable it after a few releases

Reply:
I add `merge_params_version` here to enforce backward compatibility. Integration test has some test for backward compatibility, and I also tested backward compatibility locally.

I tested in two scenarios: (1) 1 replica has new version and one replica has old version, and (2) create table in new version, then downgrade the cluster. Both are working as expected.

The key point is that deserialising replicated metadata doesn't check for EOF, so I think as long as we only add new fields to the end then it's always backward compatible.

wdyt?